File: app/src/main/java/de/blankedv/lanbahnpanel/ViewThread.java
package de.blankedv.lanbahnpanel;

import android.graphics.Canvas;
import android.view.SurfaceHolder;

import java.util.TimerTask;

public class ViewThread extends Thread {

	    private Panel mPanel;
	    private SurfaceHolder mHolder;
	    private boolean mRun = false;

	 
	    public ViewThread(Panel panel) {
	        mPanel = panel;
	        mHolder = mPanel.getHolder();
	    }
	 
	    public void setRunning(boolean run) {
	        mRun = run;
	    }
	 
	    @Override
	    public void run() {
	        while (mRun) {
                Canvas canvas = mHolder.lockCanvas();
	            if (canvas != null) {
	                mPanel.doDraw(canvas);
	                mHolder.unlockCanvasAndPost(canvas);
				 	try {
						// do not redraw more than 10 times per second
						Thread.sleep(100);
					} catch (InterruptedException e) {
						e.printStackTrace();
					}
				}
	        }
	    }

}

File: app/src/main/java/de/blankedv/lanbahnpanel/ViewThread.kt
package de.blankedv.lanbahnpanel

import android.graphics.Canvas
import android.view.SurfaceHolder

import java.util.TimerTask

class ViewThread(private val mPanel: Panel) : Thread() {
    private val mHolder: SurfaceHolder
    private var mRun = false


    init {
        mHolder = mPanel.holder
    }

    fun setRunning(run: Boolean) {
        mRun = run
    }

    override fun run() {
        while (mRun) {
            val canvas = mHolder.lockCanvas()
            if (canvas != null) {
                synchronized(mPanel.holder) {
                    mPanel.doDraw(canvas)
                }
                mHolder.unlockCanvasAndPost(canvas)

                try {
                    // max 5 frames per second
                    Thread.sleep(200)
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }

            }
        }
    }

}
